src/codeu/chat/server/Model.java

```diff
@@ -135,15 +146,133 @@ public void add(Message message) {
     return messageByText;
   }
   
+  @Override
+  public void changeAuthority(ConversationUuid conversation, Uuid targetUser, byte authorityByte){
+	StoreAccessor<Uuid, ConversationHeader> convos = this.conversationById();
+    ConversationHeader chat = convos.first(conversation.root());
+    HashMap<Uuid, Byte> accessMap = this.getPermissionMap(chat);
+    accessMap.put(targetUser, authorityByte);
+    authority.put(chat.id, accessMap);
+  }
+
+  public void initializeAuthority(Uuid conversation, Uuid targetUser, byte authorityByte)
+  {
+    StoreAccessor<Uuid, ConversationHeader> convos = this.conversationById();
+    ConversationHeader chat = convos.first(conversation);
+    HashMap<Uuid, Byte> accessMap = this.getPermissionMap(chat);
+    accessMap.put(targetUser, authorityByte);
+    authority.put(chat.id, accessMap);
+  }
+  
+  @Override
+  public boolean isMember(ConversationHeader conversation,Uuid targetUser){
+    HashMap<Uuid, Byte> accessMap = this.getPermissionMap(conversation);
+    if(!accessMap.containsKey(targetUser)){
+  	  accessMap.put(targetUser, USER_TYPE_MEMBER);
```

Comment:
This looks like a bug to me. If there's a new user who isn't a member of a conversation, checking their membership status should not make them a member.